MathProblems/836RectangleOverlap.py

```
@@ -0,0 +1,47 @@
+rec1 and rec2 represent a valid rectangle with a non-zero area.
+
+
+'''
+class Solution:
+    def isRectangleOverlap(self, rec1: list[int], rec2: list[int]) -> bool:
+          if rec1[0]>=rec2[2] or rec2[0]>=rec1[2]: return False 
+          if rec1[1]>=rec2[3] or rec2[1]>=rec1[3]: return False 
+          return True 
+class TestApp:
+      def testing_case_one(self):
+          assert Solution().isRectangleOverlap([0,0,2,2],[1,1,3,3])==True 
```

Ruff (0.11.9)

Avoid equality comparisons to `True`; use `if Solution().isRectangleOverlap([0,0,2,2],[1,1,3,3]):` for truth checks

Replace with `Solution().isRectangleOverlap([0,0,2,2],[1,1,3,3])`

(E712)